File: JavaScript/tests/Cylinder.js
import { AlgorithmEngineEx } from '../src/AlgorithmEngineEx.js';
import { Operand } from '../src/Operand.js';

// 定义圆柱信息
class Cylinder extends AlgorithmEngineEx {
  constructor(radius, height) {
    super();
    this._radius = radius;
    this._height = height;
  }

  GetParameter(parameter) {
    if (parameter === "半径") {
      return Operand.Create(this._radius);
    }
    if (parameter === "直径") {
      return Operand.Create(this._radius * 2);
    }
    if (parameter === "高") {
      return Operand.Create(this._height);
    }
    return super.GetParameter(parameter);
  }

  ExecuteDiyFunction(funcName, operands) {
    if (funcName === "求面积") {
      if (operands.length === 1) {
        const r = operands[0].ToNumber().NumberValue;
        return r * r * Math.PI;
      }
    }
    return super.ExecuteDiyFunction(funcName, operands);
  }
}

// 导出 Cylinder 类
export { Cylinder };

File: java/src/test/java/toolgood/algorithm/Tests/Cylinder.java
package toolgood.algorithm.Tests;

import java.util.List;
import toolgood.algorithm.AlgorithmEngineEx;
import toolgood.algorithm.Operand;

public class Cylinder extends AlgorithmEngineEx {
    private double _radius;
    private double _height;

    public Cylinder(double radius, double height) {
        _radius = radius;
        _height = height;
    }
    @Override
    public Operand GetParameterEx(String parameter) {
        if(parameter.equals("半径")) {
            return Operand.Create(_radius);
        }
        if(parameter.equals("直径")) {
            return Operand.Create(_radius * 2);
        }
        if(parameter.equals("高")) {
            return Operand.Create(_height);
        }
        return super.GetParameter(parameter);
    }

    @Override
    public Operand ExecuteDiyFunction(String funcName, List<Operand> operands) {
        if (funcName.equals("求面积")) {
            if (operands.size() == 1) {
                double r = operands.get(0).ToNumber("").DoubleValue();
                return Operand.Create(r * r * Math.PI);
            }
        }
        return super.ExecuteDiyFunction(funcName, operands);
    }
}